File: Leetcode/3_LSWRC.js

var LSWRC = function(s) {
    let tmp = [];
    let longest = 0;

    for (let i =0;i < s.length; i++)
    {
        const current = tmp.indexOf(s[i]);
        if (current !== -1)
        {
            tmp.splice(0, current +1);
        }

        tmp.push(s[i]);

        longest = Math.max(
            longest,
            tmp.length
        );
    }
    return longest;
    
};

console.log(LSWRC("abcabcbb"));
console.log(LSWRC("bbbbb"));
console.log(LSWRC("pwwkew"));

File: Leetcode/3_LSWRC.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        char_map = {}
        longest = 0
        start = 0
        for i, char in enumerate(s):
            if char in char_map and char_map[char] >= start:
                start = char_map[char] + 1
            char_map[char] = i
            longest = max(longest, i - start + 1)
        return longest

print(Solution().lengthOfLongestSubstring("abcabcbb"))
    